IQSS/dataverse
# OpenAPI definition endpoint

**What this PR does / why we need it**:
On Dataverse 6.0 Payara was updated and this caused the url `/openapi` to stop working and caused an exception on the server.

This PR restores the functionality by adding a Servlet on /openapi. 

We incorporated the SmallRye OpenAPI plugin on our POM (https://github.com/smallrye/smallrye-open-api/tree/main/tools/maven-plugin) which will generate files for YAML and JSON formats and deposit them on `edu/harvard/iq/dataverse/openapi/ `these files will be provided by this endpoint depending on the format requested.

**Which issue(s) this PR closes**:

- Closes #10236 
- Closes #9981

**Suggestions on how to test this**:

- check on the develop branch the url http://localhost:8080/openapi
- Update to this branch
- Check the same URL

There are additional changes to improve the quality of the OpenAPI document like changing duplicated names from Java classes.


**Is there a release notes update needed for this change?**:
Yes

***********

on Apr 17 2024: Payara has fixed the original endpoint but it is generating a spec definition that still has errors due duplicated names. We did some testing and regardless of what is on the documentation, the server is not returning the provided files on the META-INF directory, we tested this on the application root META-INF and the currently used META-INF. At this point, we decided to use this solution that will override the original endpoint and server the generated files from the smallrye plugin. 

Preview of docs: https://dataverse-guide--10328.org.readthedocs.build/en/10328/api/native-api.html#get-openapi-specification
